hilti/runtime/include/types/vector.h

```diff
@@ -66,17 +71,17 @@ class Allocator {
 
     template<class U>
     struct rebind {
-        using other = Allocator<U, Default_>;
+        using other = Allocator<U, D, Default_>;
```

Comment:
I also only made mechanical adjustments here, but using `Allocator<U, decltype(Default_), Default_>` appears equivalent and fine to me (rebind already takes care of conversion). That being said, at least to me using the straight-forward type `Allocator<U, D, Default_>` seems less surprising, but I can change this if you feel different.

Reply:
The reason I brought this up is that that way we wouldn't need to change the other places using the vector to add a 3rd template parameter on instantiation, in particular because that new parameter doesn't feel quite intuitive to me in meaning without looking at the template's definition.

Reply:
I looked again, and I think in C++17 we can get completely rid of the parameter `D` with the following: 

```cpp
template <typename T, decltype(auto) Default_> class Allocator { ...
```